engine/classes/ElggFile.php

```diff
@@ -386,6 +379,13 @@ protected function getFilestore() {
 		// need to get all filestore::* metadata because the rest are "parameters" that
 		// get passed to filestore::setParameters()
 		if ($this->guid) {
+			// check common case in a way that can hit the metadata cache.
+			if ($this->{'filestore::filestore'} === ElggDiskFilestore::class
```

Comment:
Note for later: This is such a common case that it's probably not worth keeping a filestore property for cache.